### DISCUSSION TITLE
Array and Map delta's when observer fires off.

### DISCUSSION THREAD

**Discussion by jtwigg** [mobxjs/mobx#3484] [Q&A] [answered]
I'm observing changes in Arrays and Object maps for changes. 
One the things I'd like to do is respond to the `delta` in the object that occured. It might be in the form of

```
interface Delta {
   added: []
   removed: []
   replaced: []   
}

```

I can see the Mobx context object being run during `reaction` 's and I wonder if this exists anywhere, or can it be derived?

**Any thoughts?**

**Comment by kubk** [comment, marked as answer]
So basically you'd like to see the history of changes for objects/maps? Have you tried [observe](https://mobx.js.org/intercept-and-observe.html#observe) or [spy](https://mobx.js.org/analyzing-reactivity.html#spy) ?

**Reply by jtwigg** [reply]
~~Spy was it~~ observe it was. I remember reading it in the documentation. Cheers

**Comment by mweststrate** [comment]
Note that spy is not meant for product code, only debugging ( it is not
available in production). For product logic, use observe, or deepObserve
from mobx-utils.
